[RHPAM-2684] Add NamespaceDashboard ConsoleLink

ruromero opened the pull request:
Fix https://issues.redhat.com/browse/RHPAM-2684

ruromero commented:
/assign @tchughesiv
I still need to test it on different clusters but an initial review would be great. Thanks

ruromero commented:
ready for review

ruromero commented:
/test mod

ruromero commented:
I don't understand why the `test mod` is failing. I have tried locally with and without VPN by re-cloning the branch and removing the mod cache `rm -rf $GOPATH/pkg/mod/cache/*` and I couldn't reproduce it.
Any hint?

bmozaffa commented:
/test mod

akoserwal commented:
I deployed on 4.3 cluster, help menu link is showing cluster wide.
<img width="1436" alt="Screenshot 2020-02-17 at 1 01 23 PM" src="https://user-images.githubusercontent.com/1783060/74632921-80682d80-5186-11ea-977c-f5cd7576075a.png">

ruromero commented:
> I deployed on 4.3 cluster, help menu link is showing cluster wide.
> <img alt="Screenshot 2020-02-17 at 1 01 23 PM" width="1436" src="https://user-images.githubusercontent.com/1783060/74632921-80682d80-5186-11ea-977c-f5cd7576075a.png">

@akoserwal the Help Menu Link has been created manually in the cluster by someone else for testing purposes. The one created by this code looks like this:
![2020-02-17-085104_1585x478_scrot](https://user-images.githubusercontent.com/637557/74634606-7f260900-5164-11ea-9cc6-27220ebaeefc.png)
In this case, it is a Business Central link but can be Business Central Monitoring or Decision Central, depending on the environment deployed.
Note that it requires an HTTPS link, that's why for the trial environment there shouldn't be a Launcher link.

akoserwal commented:
@ruromero: got it, thanks for the clarification

ruromero commented:
/test mod

tchughesiv commented:
@ruromero i expect that once this merges and you rebase, the mod check should pass.
https://github.com/kiegroup/kie-cloud-operator/pull/366

ruromero commented:
/test mod

tchughesiv commented on deploy/catalog_resources/redhat/1.4/businessautomation-operator.1.4.0.clusterserviceversion.yaml:
```diff
@@ -185,6 +185,18 @@ spec:
                 name: business-automation-operator
                 resources: {}
               serviceAccountName: business-automation-operator
+      clusterPermissions:
```
this new role should be persisted in `csv-gen.go` for future csv generations via `make csv` ... e.g. -
https://github.com/kiegroup/kie-cloud-operator/blob/0c5fb7968959401ddecb50a843fed895c0e4c455/tools/csv-gen/csv-gen.go#L114
https://github.com/kiegroup/kie-cloud-operator/blob/0c5fb7968959401ddecb50a843fed895c0e4c455/tools/csv-gen/csv-gen.go#L298-L299

tchughesiv commented:
@ruromero seems like the main issue is the CI platform's inability to reach the modules located at `modernc.org` ... which are all new to this PR for some reason. They aren't needed by the main module though...

tchughesiv commented:
@ruromero these mod replacements seem to solve the connectivity problem...
https://github.com/kiegroup/kie-cloud-operator/pull/370/commits/e7300a0982aee294d5c8bb4770474c12c36cd908
https://github.com/kiegroup/kie-cloud-operator/blob/e7300a0982aee294d5c8bb4770474c12c36cd908/go.mod#L79-L87

the checks are passing here - https://github.com/kiegroup/kie-cloud-operator/pull/370

let me know if you want a PR to your branch...

tnozicka commented:
/retest

tchughesiv commented:
/hold

tchughesiv commented:
@ruromero can you add a similar section as this for creating the ClusterRole yaml files?
https://github.com/kiegroup/kie-cloud-operator/blob/0c5fb7968959401ddecb50a843fed895c0e4c455/tools/csv-gen/csv-gen.go#L298-L299

ruromero commented:
> There is way too much going on here! Please separate out the operator SDK upgrade and let's make sure we know what it takes to make that happen... if we can do it right away or not.

Created this PR only with version changes: https://github.com/kiegroup/kie-cloud-operator/pull/386
Once the linked PR is merged I will rebase this one including only ConsoleLink related changes.

ruromero commented:
@bmozaffa I have rebased from #386 and added commit https://github.com/kiegroup/kie-cloud-operator/pull/365/commits/e7d5c5787c5b3f2ce9518dc8d4c979d775ed491b

tchughesiv commented:
please rebase

ruromero commented:
rebased

ruromero commented on deploy/cluster_role_binding.yaml:
```diff
@@ -0,0 +1,11 @@
+  name: kie-cloud-operator
+roleRef:
+  kind: ClusterRole
+  name: kie-cloud-operator
```
Added, thanks!

tchughesiv commented:
/hold cancel
/lgtm

bmozaffa commented:
I think we need to roll this back. While I think finalizers are a great addition to operator-utils, I don't know why reconciliation for console link should be done any differently than any other resource reconciliation we do in this operator, namely:

Based on input, in this case CR status, figure out what should be created. Use operator-utils/resource libs to create/update/delete as needed. @ruromero , any reason why that wouldn't work here?
